src/Maestro/maestro-angular/src/app/page/build-graph-table/build-graph-table.component.ts

```diff
@@ -298,6 +298,20 @@ export class BuildGraphTableComponent implements OnChanges {
     if ('graph' in changes && this.graph) {
       this.sortedBuilds = sortBuilds(this.graph);
     }
+
+    if(changes.showAllDependencies && (typeof changes.showAllDependencies.previousValue !== 'undefined') && (changes.showAllDependencies.previousValue != changes.showAllDependencies.currentValue))
+    {
+      let showSubDependenciesSetting = changes.showAllDependencies.currentValue ? "on" : "off";
+
+      this.ai.trackEvent({name: "Show Sub-Dependencies"}, {value: showSubDependenciesSetting});
```

Comment:
>Show Sub-Dependencies [](start = 33, length = 21)

I wonder if the "event" could be something more generic we can use for tracking optional features all up.

Maybe a "featureEnabled" event that has different dimension "featureName" for which feature it was, and then "featureState" for true/false

That would lead to some easy reports,
```
customEvents
| where name == "featureEnabled"
| summarize count() by customDimentions.featureName
```

Which could quickly sum up all optional features that people twiddled.

Reply:
Refactored this bit based on our conversation. Noted in the PR description what the values will be in the telemetry.